### DISCUSSION TITLE
Data contains NAs after imputation, no evidence for constant values or collinearity

### DISCUSSION THREAD

**Discussion by Michael-Covell** [amices/mice#350] [Missing data methodology]
I have NAs after I run mice() when I "skip" imputation of some variables using this line of code...
`meth[c('revenue','units','category','menu','menu_agg')] <- ""`

There is no warning message about constant values and the correlation matrix (among numeric variables) looks like this...
```
dfr <- df[,c('broilers_boosted','revenue','units')]
cor(dfr, use = "pair")

| broilers_boosted | revenue | units
1.0000000 | 0.5961082 | 0.6432247
0.5961082 | 1.0000000 | 0.6547681
0.6432247 | 0.6547681 | 1.0000000
```

I am interested only in imputing values for broilers_boosted.  Here is the full code I am running:
[Example.txt](https://github.com/amices/mice/files/5138471/Example.txt)
```
df <- read.csv("df.txt)"
ini <- mice(df, maxit = 0)
pred = ini$predictorMatrix
pred[c("revenue",'units', 'category', 'menu', 'menu_agg'),] <- 0
pred[,c('category', 'menu', 'menu_agg')] <- 0
meth <- ini$meth
meth["broilers_boosted"] <- "norm"
meth[c('revenue','units','category','menu','menu_agg')] <- "" #only part of the data is imputed when I include this line
xyz <- mice(df, predictorMatrix=pred, method=meth)
colSums(is.na(complete(xyz)))
```

Again, I am interested only in imputing values for broilers_boosted, but if (and only if) I skip imputation of the other variables, mice() only partially imputes data for broilers_boosted such that there are still NAs after imputation.  Is this a bug?

**Comment by stefvanbuuren** [comment]
This is normal behaviour. The line 
```r
meth[c('revenue','units','category','menu','menu_agg')] <- ""
```
instruct `mice` to skip imputation for some variables, but these later are needed to impute `broilers_boosted`. Hence if they contain missing values in some rows, then `mice` cannot calculate imputations for these rows, and `NA`s will result. 

You would need to remove the predictors from the imputation model for `broilers_boosted` if you do not want this from happening, e.g. by setting
```r
pred["broilers_boosted", c("revenue",'units', 'category', 'menu', 'menu_agg')] <- 0
```

**Reply by maria-jal** [reply]
I am just learning new things about multiple imputation. I also have a question related to the problem discussed here.

1.I have the same problem with missing values: I want to impute missing data from predictors with missings in the data. In my results, my imputations all have NAs even though I use other predictors besides the predictor with missings - is this also normal behavior?
2. if I would impute the missings for these predictors in my imputation process, too - would it be appropriate to exclude these imputations from my main analyses in the last step and use the predictor values from the original data set (with missings)?

I would appreciate any qualified answer.

**Reply by thomvolker** [reply]
If any of the predictors have missing values, it is not possible to create an imputation, because the fitted values cannot be calculated if any of the predictors have missing values. Using the predictorMatrix argument, you could exclude the predictors with missings, if that is something what you want. Alternatively, you can impute all variables.

The second question is hard to answer, because I don't know what your intended analysis is. Most analyses won't work if you have missing values in the predictors, and I don't see why you would want that to do something like that, but in principle you could do that, yes.

**Reply by maria-jal** [reply]
Thanks for your reply! 

In the main analyses, I would like to test for measurement invariance for a construct A across 3 groups. In the next step, I will test the association of this construct A with some criteria B and C. These criteria could be used in the imputation process as predictors but they have some missings. It feels like cheating to use imputations of these criteria in the main analyses. I am not sure which strategy would be adequate:
1. Don't use the criteria B and C as predictors and do not impute them in the imputation process
2. Use them as predictors, impute them in the imputation process and use all imputations in the main analyses
3. Like 2 but use the original variables B and C instead of the imputations in the main analyses

**Reply by gerkovink** [reply]
`It feels like cheating to use imputations of these criteria in the main analyses.`

Can you elaborate on why you feel this is the case? I would argue that not modeling the uncertainty associated with the missingness is cheating. After all, you're more than happy to model the analyses, but you would be hesitant to use the same or similar modeling techniques to infer about the missingness problem? There is a nice discussion of this *persistent thought* in [Aboyami, Gelman and Levy (2011)](http://www.stat.columbia.edu/~gelman/bayescomputation/abayomi.pdf)

So, I would do option 2

**Reply by stefvanbuuren** [reply]
Pros en cons of options:

1. Not recommended because omission of B and C introduces a bias towards zero in the association analysis.
2. Unbiases w.r.t. association analysis. Uses all cases. Take care to allow for differences between the A group during imputation, e.g. by including an additive or interaction term, or by imputing in three groups separately.
3. As 2, but with reduced power.

**Reply by maria-jal** [reply]
Thank you all for your replies!
